### ISSUE TITLE
How to perform checks after expect().toBeObservable() ?

### ISSUE THREAD

**Issue by Spongeroberto** [just-jeb/jest-marbles#141]
Back when I was using karma with jasmine-marbles I could write tests like this:

```javascript
expect(result).toBeObservable(expected);
expect(mockInsideObservable).toHaveBeenCalled();
```
Which is easy to read and write. The main thing here is that expect.toBeObservable acts synchronously and waits for all the code inside the observable to be executed.

The same code in jest and jest-marbles will always fail with the following message:

> Expected mock function to have been called, but it was not called

Because the last assertion is evaluated before the observable. Is this expected behavior? If so, would it be possible to provide some sort of promise or callback to the observable assertion?

**Comment by just-jeb** [comment]
I'm pretty sure that you can subscribe to the observable and inside the subscription function do the assertion. Let me know if it works for you.

**Comment by just-jeb** [comment]
Currently it works in a way that the scheduler is being flashed _after_ each test, therefore your observable is essentially invoked in [`afterEach`](https://github.com/just-jeb/jest-marbles/blob/master/index.ts#L67).  
I agree it should change and work in the same way `jasmine-marbles` work.  
Until it is implemented please use the workaround provided above.

**Comment by nvdbleek** [comment]
Hi, I also need to do a check if a mock is called with the correct parameters. But I can't get is to work. My approach was to add the following to the test (result$ is the result stream of my epic):
```result$.subscribe({
      complete: () => {
        console.error((uploadFile as jest.Mock<any>).mock.calls);
        expect(uploadFile).toBeCalledWith(destinationURI, file);
      },
    });
```

But the test never fails event adding `expect(true).toBeFalsy();` in the complete handler won't fail the test. What I'm a doing wrong?

**Comment by just-jeb** [comment]
@nvdbleek Can you please post your whole test?

**Comment by just-jeb** [comment]
I am gonna close this in favor of #112 (essentially a duplicate). Please follow on that issue to get updates.

**Comment by just-jeb** [comment]
Please use [`toSatisfyOnFlush`](https://github.com/just-jeb/jest-marbles#tosatisfyonflush), available since 2.5.0.  
Let me know if that helps.

**Comment by nicos68** [comment]
> Please use [`toSatisfyOnFlush`](https://github.com/just-jeb/jest-marbles#tosatisfyonflush), available since 2.5.0. Let me know if that helps.

Hi,

so that would mean running the observable twice if I need to test both its final value and the spy's call arguments (as is the case in the original message of this issue). How can I do the same with running the observable only once ?

**Comment by just-jeb** [comment]
You do the spy checks inside the `satisfyOnFlush`, the observable runs only once (essentially when the TestScheduler is flushed).
